dapp/src/components/buySell/ApproveSwap.js

```diff
@@ -82,7 +104,6 @@ const ApproveSwap = ({
         return
       }
     }
-    setCoinApproved(false)
     setStage('approve')
```

Comment:
@rolandpo curious was this bit of code below a UX requirement? (it might be I am really not sure...)

My first hunch would be to show the button displayed all of the time until the transaction is not mined. But I could be wrong...